packages/expo-cli/src/commands/build-native/index.ts

```diff
@@ -0,0 +1,44 @@
+import { Platform, BuildType } from '@expo/build-tools';
+import { UserManager, User } from '@expo/xdl';
+import { Command } from 'commander';
+
+import log from '../../log';
+import BaseBuilder, { StatusResult } from './BaseBuilder';
+import { wait } from './utils';
+
+export type Options = {
+  platform: Platform;
+  type: BuildType;
+}
+
+
+async function buildAction(projectDir: string, options: Options) {
+  const user: User = await UserManager.ensureLoggedInAsync();
+  const builder = new BaseBuilder(projectDir, user, options);
```

Comment:
the name suggests that this is a base class and there exist ios builder and android builder